vdr/didx509/resolver.go

```diff
@@ -119,6 +119,10 @@ func (r Resolver) Resolve(id did.DID, metadata *resolver.ResolveMetadata) (*did.
 	if bytes.Equal(caFingerprintCert.Raw, validationCert.Raw) {
 		return nil, nil, fmt.Errorf("did:x509 ca-fingerprint refers to leaf certificate, must be either root or intermediate CA certificate")
 	}
+	leafCert := chain[0]
+	if !bytes.Equal(leafCert.Raw, validationCert.Raw) {
+		return nil, nil, fmt.Errorf("did:x509 x5c leaf certificate does not match x5t or x5t#S256 header; x5c header must be sorted from leaf- to root certificate")
```

Comment:
how does this relate to the `x5t#S256` header? We also do not allow `x5t` since it uses sha-1